traces/traces.go

```diff
@@ -0,0 +1,77 @@
+package traces
+
+import (
+	"context"
+	"fmt"
+	"runtime/debug"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const instrumentationPkg = "github.com/osquery/osquery-go"
+
+var internalVersion string
+
+// osqueryGoVersion returns the version of osquery-go, to be used to set the instrumentation
+// version `otel.scope.version`. It looks through build info to determine the current version
+// of the osquery-go package. Once determined, it saves the version to avoid performing this
+// work again. If the version cannot be determined, the version defaults to the current version,
+// which is hardcoded.
+func osqueryGoVersion() string {
+	if internalVersion != "" {
+		return internalVersion
+	}
+	if info, ok := debug.ReadBuildInfo(); ok {
+		for _, dep := range info.Deps {
+			if dep == nil {
+				continue
+			}
+			if dep.Path == instrumentationPkg {
+				internalVersion = dep.Version
+				return internalVersion
+			}
+		}
+	}
+
+	// Couldn't get the version from runtime build info -- save and return 0.0.0,
+	// which is the current osquery-go version.
+	internalVersion = "0.0.0"
+	return internalVersion
+}
+
+// By default, use the global tracer provider
```

Comment:
Could we add a comment that this "global tracer provider" is no-op by default?

Reply:
Yes absolutely -- updated!